ports/stm32/rfcore.c

```diff
@@ -643,9 +647,9 @@ void rfcore_ble_hci_cmd(size_t len, const uint8_t *src) {
     LL_C1_IPCC_SetFlag_CHx(IPCC, ch);
 }
 
-void rfcore_ble_check_msg(int (*cb)(void *, const uint8_t *, size_t), void *env) {
-    parse_hci_info_t parse = { cb, env, false };
-    tl_check_msg(&ipcc_mem_ble_evt_queue, IPCC_CH_BLE, &parse);
+size_t rfcore_ble_check_msg(rfcore_ble_msg_callback_t cb, mp_bluetooth_hci_uart_readchar_t handler) {
```

Comment:
I think I'd prefer to keep this signature as-is (except it returns a size_t now), because it allows for a generic callback with a generic environment variable.  With the change here to force the env var to be a `mp_bluetooth_hci_uart_readchar_t` it's making it more restrictive what can be passed in, without any real benefit.

@jimmo what do you think?

Reply:
I think using `rfcore_ble_msg_callback_t` for the first arg is a definite improvement. For the second argument, I generally prefer to avoid generic arguments wherever possible (and do whatever is necessary when the need arises) but yeah no strong feelings either way here. I guess in this case it's unnecessary coupling between bthciport.c and rfcore.c.

Reply:
Yes I think it's an unnecessary coupling.

If it must be a `mp_bluetooth_hci_uart_readchar_t` then rfcore may as well just call that readchar function directly, instead of going via `cb`.

So we can keep the `rfcore_ble_msg_callback_t` but I think revert `mp_bluetooth_hci_uart_readchar_t handler` to `void *env`.

Reply:
Done.